src/launchpad/size/analyzers/apple.py

```diff
@@ -553,38 +554,41 @@ def _test_strip_symbols_removal(self, parser: MachOParser, binary_path: Path) ->
                 return 0
 
             # Create a copy of the binary and remove the symbols to see the new size
-            binary_copy = lief.MachO.parse(str(binary_path))  # type: ignore
-            if not binary_copy or binary_copy.size == 0:
-                logger.warning("Failed to create binary copy for symbol removal testing")
-                return 0
+            with trace_ctx("strip_symbols.create_binary_copy"):
+                binary_copy = lief.MachO.parse(str(binary_path))  # type: ignore
+                if not binary_copy or binary_copy.size == 0:
+                    logger.warning("Failed to create binary copy for symbol removal testing")
+                    return 0
 
-            binary_copy_obj = binary_copy.at(0)
+                binary_copy_obj = binary_copy.at(0)
 
-            removed_count = 0
-            for symbol in removable_symbols:
-                if symbol.name:
-                    binary_copy_obj.remove_symbol(str(symbol.name))
-                    removed_count += 1
+            with trace_ctx("strip_symbols.remove_symbols"):
+                removed_count = 0
+                for symbol in removable_symbols:
+                    if parser.binary.can_remove(symbol):
```

Comment:
Pushing the `can_remove()` call down here + avoiding a second string conversion for `str(symbol.name)` are the main changes.